test/test-debuglet.ts

```diff
@@ -289,7 +289,7 @@ describe('Debuglet', function() {
   });
 
   describe('setup', function() {
-    before(function() { oldGP = process.env.GCLOUD_PROJECT; });
+    before(function() { oldGP = String(process.env.GCLOUD_PROJECT); });
```

Comment:
If `process.env.GCLOUD_PROJECT` is `undefined`, then it looks like `oldGP` will be set to the string `'undefined'` which could cause reseting the value of `process.env.GCLOUD_PROJECT` later to not work as expected.

Reply:
We don't yet have style best practices for typescript but I think using Typescript casts (`process.env.GCLOUD_PROJECT as string`) instead of Javascript coersions (`String(process.env.GCLOUD_PROJECT)`) is preferable just so that the actual behavior at runtime is not changed (especially to avoid bugs such as the one @DominicKramer described).

Reply:
Redefine oldGP variable to be string|undefined in test-debuglet.ts